release/src/version_update/main.py

```diff
@@ -0,0 +1,173 @@
+import urllib.request
+import json
+import sys
+import os
+from retry import retry
+
+HTTP_REQUEST_RETRIES = 3
+HTTP_REQUEST_DELAY_IN_SECONDS = 2
+HTTP_REQUEST_DELAY_MULTIPLIER = 2
+packageUser =  os.environ["packageUser"]
+packagePAT = os.environ["packagePAT"]
+packageEmail =  os.environ["packageEmail"]
+organization = 'ballerina-platform'
+
+def main():
+    print("Checking Ballerina Distribution for stdlib version updates")
+    moduleList = getStdlibModules()
+    repo = configureGithubRepository()
+    propertiesFile = fetchPropertiesFile(repo)
+    modifiedPropertiesFile, commitFlag, updatedModules = updatePropertiesFile(propertiesFile, moduleList)
+    if commitFlag:
+        commitChanges(modifiedPropertiesFile, repo, updatedModules)
+        createPullRequest(repo)
+        print("Updated gradle.properties file in Ballerina Distribution Successfully")
+    else:
+        print("Stdlib versions in gradle.properties file are up to date")
+
+# Get stdlib module details from stdlib_modules.json file
+def getStdlibModules():
+    try:
+        with open('./release/resources/stdlib_modules.json') as f:
+            moduleList = json.load(f)
+    except:
+        print('Failed to read stdlib_modules.json')
+        sys.exit()
+
+    return moduleList['modules']
+
+# Fetch ballerina-distribution repository with GitHub credentials
+def configureGithubRepository():
+    g = Github(packagePAT)
+    try:
+        repo = g.get_repo(organization + "/" + 'ballerina-distribution')
+    except:
+        print("Error fetching repository ballerina-distribution")
+
+    return repo
+
+# Fetch the gradle.properties file from the ballerina-distribution repo
+def fetchPropertiesFile(repo):
+    try:
+        branch = repo.get_branch(branch="automated-stdlib-version-update")
+        file = repo.get_contents("gradle.properties", ref="automated-stdlib-version-update")
+    except GithubException:
+        file = repo.get_contents("gradle.properties")
+
+    data = file.decoded_content.decode("utf-8")
+
+    return data
+
+# Update stdlib module versions in the gradle.properties file with module details fetched from stdlib_modules.json
+def updatePropertiesFile(data, modules):
+    modifiedData = ''
+    updatedModules = []
+    pointer = ''
```

Comment:
What does this `pointer` do? The name is ambiguous.

Reply:
It points to a specific line in `gradle.properties`

Reply:
Then maybe `currentLine` would be a preferable name. Shall we rename it?

Reply:
Updated